File: 1200. Minimum Absolute Difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()

        n = len(arr)
        min_abs = 10**10
        for i in range(n - 1):
            min_abs = min(min_abs, abs(arr[i] - arr[i + 1]))
        
        ans = []
        for i in range(n - 1):
            if abs(arr[i] - arr[i + 1]) == min_abs:
                ans.append([arr[i], arr[i + 1]])
        return ans

File: Sorting/1200. Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& a) 
    {
        sort(a.begin(), a.end());

        int n = a.size();
        int minABS = INT_MAX;
        for(int i = 1; i < n; i++)
        {
            if(minABS > abs(a[i - 1] - a[i])) 
                minABS = abs(a[i - 1] - a[i]);
        }

        vector<vector<int>> ans;
        for(int i = 1; i < n; i++)
        {
            if(abs(a[i - 1] - a[i]) == minABS) 
                ans.push_back({a[i - 1], a[i]});
        }
        return ans;
    }
};